lib/widgets/theme.dart

```diff
@@ -0,0 +1,80 @@
+import 'package:flutter/material.dart';
+
+import 'text.dart';
+
+ThemeData outerThemeData(BuildContext context) {
```

Comment:
Is there a coherent description we can give for what the meaning of this "outer ThemeData" is?

It seems like it's basically an intermediate step in the calculation of the "inner ThemeData", which is the theme data we actually want to use in the app.

In that case, can we have one function that just computes the desired ThemeData start to finish, from a single BuildContext? I think that'd make it easier to reason about how the caller should use the function. That function could even have this intermediate step as a ThemeData local variable, if that's helpful.